src/common/queries/credits.ts

```diff
@@ -16,25 +16,46 @@ import { route } from '$app/common/helpers/route';
 import { useAtomValue } from 'jotai';
 import { invalidationQueryAtom } from '$app/common/atoms/data-table';
 
-export function useBulkAction() {
+const successMessages = {
+  email: 'emailed_credits',
+  mark_sent: 'marked_credit_as_sent',
+};
+
+interface Params {
+  onActionCall?: () => void;
+}
+export const useBulk = (params?: Params) => {
   const queryClient = useQueryClient();
   const invalidateQueryValue = useAtomValue(invalidationQueryAtom);
 
-  return (id: string, action: 'archive' | 'restore' | 'delete') => {
+  const { onActionCall } = params || {};
+
+  return (
+    ids: string[],
+    action: 'archive' | 'restore' | 'delete' | 'email' | 'mark_sent'
+  ) => {
     toast.processing();
 
     request('POST', endpoint('/api/v1/credits/bulk'), {
       action,
-      ids: [id],
+      ids,
     })
-      .then(() => toast.success(`${action}d_credit`))
-      .finally(() => {
+      .then(() => {
+        const message =
+          successMessages[action as keyof typeof successMessages] ||
+          `${action}d_invoice`;
+
+        toast.success(message);
+
         queryClient.invalidateQueries('/api/v1/credits');
 
-        queryClient.invalidateQueries(route('/api/v1/credits/:id', { id }));
+        ids.forEach((id) =>
+          queryClient.invalidateQueries(route('/api/v1/credits/:id', { id }))
+        );
 
         invalidateQueryValue &&
           queryClient.invalidateQueries([invalidateQueryValue]);
-      });
+      })
+      .finally(() => onActionCall?.());
```

Comment:
Isn't this old pattern we decided not to go with?

Reply:
@beganovich Ah, sure. Actually, it was only used to close the modal. However, the solution has now been updated to avoid using this pattern. Let me know your thoughts.